rpc_server.go

```diff
@@ -145,10 +184,13 @@ func (s *SourceControl) broadcastTriggerState() {
 	}
 }
 
-// SendAllStatus sends all relevant status messages to
+// SendAllStatus causes a broadcast to clients containing all broadcastable status info
 func (s *SourceControl) SendAllStatus(dummy *string, reply *bool) error {
+	log.Println("A Client has requested to send all status")
+	s.broadcastTriggerState()
 	s.broadcastUpdate()
-	s.clientUpdates <- ClientUpdate{"SENDALL", 0}
+	s.broadcastTriggerState()
+	*reply = true
 	return nil
 }
 
```

Comment:
It looks to me like you are reverting to an earlier version of this method. I'm okay with your comment, but I think the function body should read 
```go
func (s *SourceControl) SendAllStatus(dummy *string, reply *bool) error {
	s.broadcastUpdate()
	s.clientUpdates <- ClientUpdate{"SENDALL", 0}
	return nil
}
```